internal/v6provider/resource_load_balancer_rule.go

```diff
@@ -0,0 +1,762 @@
+package v6provider
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/krystal/go-katapult"
+	"github.com/krystal/go-katapult/core"
+)
+
+type (
+	LoadBalancerRuleResource struct {
+		M *Meta
+	}
+
+	LoadBalancerRuleResourceModel struct {
+		ID                types.String `tfsdk:"id"`
+		LoadBalancerID    types.String `tfsdk:"load_balancer_id"`
+		Algorithm         types.String `tfsdk:"algorithm"`
+		DestinationPort   types.Int64  `tfsdk:"destination_port"`
+		ListenPort        types.Int64  `tfsdk:"listen_port"`
+		Protocol          types.String `tfsdk:"protocol"`
+		ProxyProtocol     types.Bool   `tfsdk:"proxy_protocol"`
+		Certificates      types.List   `tfsdk:"certificates"`
```

Comment:
This should also be a set, as the ordering is not guaranteed server-side.

And looking into the API a bit further, we should probably turn this into `CertificateIDs`/`certificate_ids` turning it into a set of strings. It's meant for associating existing certificates to a rule.

```suggestion
		Certificates      types.Set    `tfsdk:"certificate_ids"`
```

The way it is right now, the schema sets `name` as required as well, despite it never using it when creating/updating the rule. Hence just a string for the ID I think is more than enough.